### DISCUSSION TITLE
@Dependency(\.dismiss) not working under a deep-linking scenario

### DISCUSSION THREAD

**Discussion by wltrup** [pointfreeco/swift-composable-architecture#2293] [Q&A]
Hello :)

We have a situation where `@Dependency(\.dismiss)` works fine if scenes are navigated into by the user but not if navigated into through deep-linking. Dismissing by using `@Environment(\.dismiss)` works in both cases. I built a small toy app (attached) to demonstrate.

In the screen recording, you see that deep-linking from the root to A to B then tapping the dismiss button that uses  `@Environment(\.dismiss)` works in both B then A. But deep-linking from the root to A to B then tapping the dismiss button that uses `@ Dependency(\.dismiss)`, in B, does not work.

Yet, navigating from the root to A then to B one scene at a time (tapping the "Show Next Scene" button), then tapping either dismiss button works fine in both B and A (this isn't shown in the screen recording, to save on space). 

Thanks for taking the time to take a look.

[YANTP.zip](https://github.com/pointfreeco/swift-composable-architecture/files/12054002/YANTP.zip)

https://github.com/pointfreeco/swift-composable-architecture/assets/1015142/4e850a2a-75dd-407e-9b98-b9d7f7825d72

**Comment by stephencelis** [comment]
Haven't had a chance to dive in too deeply, but this looks like it may be a SwiftUI bug with `navigationDestination` bindings. If you change each `navigationDestination` view modifier to a `sheet` modifier, it all works just fine. SwiftUI's `navigationDestination(isPresented:)` view modifier is sadly buggier than other presentation APIs, and so you may need to resort to workarounds like using `@Environment(\.dismiss)` instead.
